include/albatross/src/core/linear_combination.hpp

```diff
@@ -74,6 +76,16 @@ inline auto difference(const X &x, const Y &y) {
   return difference(vx, vy);
 }
 
+template <typename X> inline auto to_linear_combination(const X &x) {
```

Comment:
Why make this a free function rather than a constructor of `LinearCombination`?

Reply:
One reason is type inference. If, for example, you have some type:
```
SomeWrapper<variant<Foo, Bar>, WowThatsALongTypeName> x;
```
If you want to create a linear combination with that type using a constructor I think you'd have to do:
```
LinearCombination<SomeWrapper<variant<Foo, Bar>, WowThatsALongTypeName>> combo(x);
```
this function let's you do:
```
const auto combo = to_linear_combination(x);
```

Reply:
I see.  What I envisioned only works in C++17 and later.